### PR TITLE
docs: update broken links for apm-guide-ref #1385

### PR DESCRIPTION
**Pull request by ollyhowell**
Links updated as per https://github.com/elastic/observability-docs/issues/1385. The old links will break when 7.17 releases. This PR and all backports must be merged before the 7.17 release. Thanks!

### Backports

- [ ] 3.x: https://github.com/elastic/apm-agent-nodejs/pull/2542

### CODE REVIEW

File: docs/agent-api.asciidoc [hunk 1]
```diff
@@ -104,7 +104,7 @@ containing the data about to be sent to the APM Server.
 
 The format of the payload depends on the event type being sent.
 For details about the different formats,
-see the {apm-server-ref-v}/intake-api.html[APM Server intake API documentation].
+see the {apm-guide-ref-v}/intake-api.html[APM Server intake API documentation].
```
Comment (bmorelli25) [inline docs/agent-api.asciidoc:107]:
```suggestion
see the {apm-guide-ref}/intake-api.html[APM Server intake API documentation].
```

File: docs/agent-api.asciidoc [hunk 2]
```diff
@@ -164,7 +164,7 @@ but the `fn` will only be called with span payloads.
 [small]#Added in: v3.14.0#
 
 Use `addMetadataFilter(fn)` to supply a filter function for the
-{apm-server-ref-v}/metadata-api.html[metadata object]
+{apm-guide-ref-v}/metadata-api.html[metadata object]
```
Comment (bmorelli25) [inline docs/agent-api.asciidoc:167]:
```suggestion
{apm-guide-ref}/metadata-api.html[metadata object]
```

File: docs/agent-api.asciidoc [hunk 3]
```diff
@@ -184,7 +184,7 @@ apm.addMetadataFilter(function dropArgv(metadata) {
 
 Warning: It is the responsibility of the author to ensure the returned object
 conforms to the
-{apm-server-ref-v}/metadata-api.html#metadata-schema[metadata schema]
+{apm-guide-ref-v}/metadata-api.html#metadata-schema[metadata schema]
```
Comment (bmorelli25) [inline docs/agent-api.asciidoc:187]:
```suggestion
{apm-guide-ref}/metadata-api.html#metadata-schema[metadata schema]
```

File: docs/agent-api.asciidoc [hunk 4]
```diff
@@ -403,7 +403,7 @@ To ease debugging it's possible to send some extra data with each error you send
 The APM Server intake API supports a lot of different metadata fields,
 most of which are automatically managed by the Elastic APM Node.js Agent.
 But if you wish you can supply some extra details using `user` or `custom`.
-For more details on the properties accepted by the events intake API see the {apm-server-ref-v}/events-api.html[events intake API docs].
+For more details on the properties accepted by the events intake API see the {apm-guide-ref-v}/events-api.html[events intake API docs].
```
Comment (bmorelli25) [inline docs/agent-api.asciidoc:406]:
```suggestion
For more details on the properties accepted by the events intake API see the {apm-guide-ref}/events-api.html[events intake API docs].
```

File: docs/intro.asciidoc [hunk 5]
```diff
@@ -30,6 +30,6 @@ You can then use the APM app in Kibana to gain insight into latency issues and e
 [[additional-components]]
 === Additional Components
 
-APM Agents work in conjunction with the {apm-server-ref-v}/index.html[APM Server], {ref}/index.html[Elasticsearch], and {kibana-ref}/index.html[Kibana].
-The {apm-overview-ref-v}/index.html[APM Overview] provides details on how these components work together,
-and provides a matrix outlining {apm-overview-ref-v}/agent-server-compatibility.html[Agent and Server compatibility].
+APM Agents work in conjunction with the {apm-guide-ref-v}/index.html[APM Server], {ref}/index.html[Elasticsearch], and {kibana-ref}/index.html[Kibana].
```
Comment (bmorelli25) [inline docs/intro.asciidoc:33]:
```suggestion
APM Agents work in conjunction with the {apm-guide-ref}/index.html[APM Server], {ref}/index.html[Elasticsearch], and {kibana-ref}/index.html[Kibana].
```

File: docs/intro.asciidoc [hunk 6]
```diff
@@ -30,6 +30,6 @@ You can then use the APM app in Kibana to gain insight into latency issues and e
...
+The {apm-guide-ref}/index.html[APM Overview] provides details on how these components work together,
```
Comment (bmorelli25) [inline docs/intro.asciidoc:34]:
```suggestion
The {apm-guide-ref}/index.html[APM Guide] provides details on how these components work together,
```

File: docs/supported-technologies.asciidoc [hunk 7]
```diff
@@ -22,7 +22,7 @@ that not all core Node.js API's can be instrumented without the use of Async Hoo
 [[elastic-stack-compatibility]]
 === Elastic Stack Compatibility
 
-This agent is compatible with {apm-server-ref-v}[APM Server] v6.5 and above.
+This agent is compatible with {apm-guide-ref-v}[APM Server] v6.5 and above.
```
Comment (bmorelli25) [inline docs/supported-technologies.asciidoc:25]:
```suggestion
This agent is compatible with {apm-guide-ref}[APM Server] v6.5 and above.
```

File: docs/troubleshooting.asciidoc [hunk 8]
```diff
@@ -86,7 +86,7 @@ APM Server transport error (ECONNREFUSED): connect ECONNREFUSED 127.0.0.1:8200
 These may indicate an issue with the agent configuration (see <<server-url>>,
 <<secret-token>> or <<api-key>>), a network problem between agent and server, or
 that the APM server is down or misconfigured (see
-{apm-server-ref-v}/troubleshooting.html[the APM server troubleshooting docs]).
+{apm-guide-ref-v}/troubleshooting.html[the APM server troubleshooting docs]).
```
Comment (bmorelli25) [inline docs/troubleshooting.asciidoc:89]:
```suggestion
{apm-guide-ref}/troubleshooting.html[the APM server troubleshooting docs]).
```

**Review by bmorelli25 — APPROVED** [review]
Thanks, Olly! LG. I'll tag the Node.js team for review and create the lone backport.

File: CHANGELOG.asciidoc [hunk 9]
```diff
@@ -373,7 +373,7 @@ still supported. Users should switch to using `longFieldMaxLength`. If
 now use the `longFieldMaxLength` value.
 +
 Note that ultimately the maximum length of any tracing field is limited by the
-{apm-server-ref-v}/configuration-process.html#max_event_size[`max_event_size`]
+{apm-guide-ref}/configuration-process.html#max_event_size[`max_event_size`]
```
Comment (trentm) [inline CHANGELOG.asciidoc:376]: https://github.com/elastic/observability-docs/issues/1385 suggests this one should be "Remove from Node.js Agent"?

File: docs/agent-api.asciidoc [hunk 10]
```diff
@@ -104,7 +104,7 @@ containing the data about to be sent to the APM Server.
 
 The format of the payload depends on the event type being sent.
 For details about the different formats,
-see the {apm-server-ref-v}/intake-api.html[APM Server intake API documentation].
+see the {apm-guide-ref}/intake-api.html[APM Server intake API documentation].
```
Comment (trentm) [inline docs/agent-api.asciidoc:107]: https://github.com/elastic/observability-docs/issues/1385 says

`{apm-server-ref}/intake-api.html` -> `{apm-guide-ref}/api.html`

Though perhaps for this one we'd (eventually?) want https://www.elastic.co/guide/en/apm/guide/7.17/api-events.html ?  I'm not sure if the intent of this PR is to just fix the `{apm-server-ref-v}` reference, or to also change from the "Legacy APM Server Reference" to the new Agent-based APM server docs.

**Review by trentm — APPROVED** [review]

File: CHANGELOG.asciidoc (hunk 9 above)
Comment (bmorelli25) [reply CHANGELOG.asciidoc:376]: That was an incorrect assumption on my part. This config variable is still valid for the standalone method of running APM Server. We'll need to remove this at a later date.

File: docs/agent-api.asciidoc (hunk 10 above)
Comment (bmorelli25) [reply docs/agent-api.asciidoc:107]: Unfortunately, the goal here is just to fix the outdated `{apm-server-ref-v}` refs. There are improvements that could be made to all of the agent docs to better differentiate between running APM Server in standalone mode vs the APM integration, but I don't have the bandwidth to tackle those right now.

As for the API link. It looks like this was a mistake, but I'd agree that the current link is better than what I had proposed in elastic/observability-docs#1385.

**Pull request merged**